### DISCUSSION TITLE
Help using service-flake with node app that requires redis service load before nginx service

### DISCUSSION THREAD

**Discussion by ParetoOptimalDev** [juspay/services-flake#48] [General]
This is mostly just a description of what I'm doing and I'll continue to post questions and/or commentary of my experience along the way, but I do have one question (that I'll attempt to answer):

Does service-flake support ordering processes?

answer: It [appears yes](https://github.com/juspay/services-flake/blob/main/example/flake.nix#L64):

```
            settings.processes.test = {
              command = pkgs.writeShellApplication {
                name = "pg1-test";
                runtimeInputs = [ config.services.postgres.pg1.package ];
                text = ''
                  echo 'SELECT version();' | psql -h 127.0.0.1 ${dbName}
                '';
              };
              depends_on."pg1".condition = "process_healthy";
            };
```

Here is the app that is currently failing that uses devenv:

https://github.com/ParetoOptimalDev/reproduce--devenv-nginxhostctl-node-issue

I see that redis is supported, but nginx isn't. It looks pretty easy to add a new service though and I think I can do it.

**Comment by srid** [comment]
> I see that redis is supported, but nginx isn't. It looks pretty easy to add a new service though and I think I can do it

Would be happy to see a PR. 👍🏿 

Generally, we just copy over devenv's module and then iterate from there.

**Reply by srid** [reply]
> It looks pretty easy to add a new service though

❤️  (credits to @shivaraj-bh)
